app/lib/actions/katello/content_view/publish.rb

```diff
@@ -23,10 +23,17 @@ def plan(content_view, description = "")
           sequence do
             plan_action(ContentView::AddToEnvironment, version, library)
             concurrence do
-              content_view.repositories_to_publish_by_library_instance.each do |_library_instance, repositories|
+              content_view.repositories_to_publish_by_library_instance.each do |library_instance, repositories|
```

Comment:
I wonder if repositories_to_publish_by_library_instance could be changed so that it produces the same result across both scenarios (not grouping by nil library_instance).  Possibly using a block to do this